apps/settings/js/modules/mvvm/observable.js

```diff
@@ -1,102 +1,219 @@
-'use strict';
-
-define(function() {
-  function unobserve(_eventHandlers, prop, handler) {
-    // arguments in reverse order to support .bind(handler) for the
-    // unbind from all case
-    function removeHandler(handler, prop) {
-      var handlers = _eventHandlers[prop];
-      if (!handlers) {
+define(function(require) {
+  'use strict';
+
+  var DependencyMap = require('modules/base/dependency_map');
+
+  // Observable
+  var OP_PREFIX = (name) => { return '$OP_' + name; };
+
+  function Observable(obj) {
+    this._init(obj);
+  }
+
+  Observable.prototype._init = function o_init(obj) {
+    if (!obj) {
+      return;
+    }
+    for (var name in obj) {
+      // If name is a function, simply add it to the observable.
+      if (typeof obj[name] === 'function') {
+        this[name] = obj[name];
+      } else {
+        _defineObservableProperty(this, name, {
+          value: obj[name]
+        });
+      }
+    }
+  };
+
+  Observable.prototype._notify = function o__notify(name, newValue, oldValue) {
+    var observers = !!this._observers && this._observers[name];
+    if (observers) {
+      observers.forEach(function(observer) {
+        observer(newValue, oldValue);
+      });
+    }
+  };
+
+  Observable.prototype._removeObserver =
+    function o__removeObserver(observer, name) {
+      // arguments in reverse order to support .bind(observer) for the
+      // unbind from all case
+      var observers = !!this._observers && this._observers[name];
+      if (!observers) {
         return;
       }
-      var index = handlers.indexOf(handler);
+      var index = observers.indexOf(observer);
       if (index >= 0) {
-        handlers.splice(index, 1);
+        observers.splice(index, 1);
       }
+  };
+
+  Observable.prototype.observe = function o_observe(name, observer) {
+    var propertyInfo =
+      this._observablePropertiesInfo && this._observablePropertiesInfo[name];
+    if (!propertyInfo) {
+      throw new Error('Observable: property ' + name + ' does not exist');
```

Comment:
I think we can directly `console.log` the error here instead of throwing them away directly because by doing so, we can still make our code workable even if one day someone accidentally remove the `observablePropertiesInfor[name]`